driver/main.c

```diff
@@ -1721,6 +1627,7 @@ static inline int drop_event(struct ppm_consumer_t *consumer,
 			if (consumer->is_dropping == 0) {
 				consumer->is_dropping = 1;
 				record_drop_e(consumer, ns, drop_flags);
+				return 0;
```

Comment:
Bpf driver was returning false here, and it seems legit (ie: record a `DROP_E` event and don't drop it).
Note: this is the same that was already being done for `record_drop_x` below.

Reply:
Have you tested it? I'm somewhat worried the `DROP_E` event will end up in the middle of the original event (but will probably get overwritten by it too).

Effectively, the (implied) `return 1` increased the sampling window size by one event in each direction. This is only affecting up to two non-UF_NEVER_DROP/UF_ALWAYS_DROP events per second, but even then, I'd rather have an extra dropped event than a corrupted ring buffer (or a dropped DROP_E/DROP_X).

Reply:
I was not seeing any corruption; but i can try again with a long lasting test.

Reply:
This isn't going to be easy to reproduce without custom code (AFAIK the commercial agent is the only user of the sampling feature right now). You'll have to set the sampling ratio manually (e.g. to 2) and then (apart from any corruption we may observe; not sure how to detect that automatically) ensure we're getting one PPM_DROP_E around subsecond offset 500ms (i.e. halfway through each second) and a PPM_DROP_X around subsecond offset 0ms (i.e. the start of each second).

Reply:
I see; i think we might as well revert this tiny line :D